File: 1667-find-kth-bit-in-nth-binary-string/find-kth-bit-in-nth-binary-string.py
class Solution:
    def invert(self, s: str) -> str:
        return ''.join('1' if c == '0' else '0' for c in s)

    def reverse(self, s: str) -> str:
        return s[::-1]


    def findKthBit(self, n: int, k: int) -> str:
        s = "0"
        
        for i in range(1, n):
            inverted = self.invert(s)
            reversed_inverted = self.reverse(inverted)
            s = s + "1" + reversed_inverted

        return s[k - 1]

File: 1667-find-kth-bit-in-nth-binary-string/find-kth-bit-in-nth-binary-string.cpp
class Solution {
public:
    string invert(string s) {
        for (char &c : s) {
            c = (c == '0') ? '1' : '0';
        }
        return s;
    }

    string reverse(string s) {
        std::reverse(s.begin(), s.end());
        return s;
    }
    
    char findKthBit(int n, int k) {
        string s = "0";
        for(int i=1;i<n;i++){
            string inverted = invert(s);
            string reversedInverted = reverse(inverted);
            s = s + "1" + reversedInverted;
        }
        return s[k-1];
    }
};